wpilib/wpilib/command/scheduler.py

```diff
@@ -217,52 +217,31 @@ def enable(self):
         """
         self.disabled = False
 
-    def getName(self):
-        return "Scheduler"
-
-    def getType(self):
-        return "Scheduler"
-
-    def initTable(self, subtable):
-        self.commands = []
-        self.ids = []
-        self.toCancel = []
-
-        if subtable is not None:
-            self.namesEntry = subtable.getEntry("Names")
-            self.idsEntry = subtable.getEntry("Ids")
-            self.cancelEntry = subtable.getEntry("Cancel")
-
-            self.namesEntry.setStringArray(self.commands)
-            self.idsEntry.setDoubleArray(self.ids)
-            self.cancelEntry.setDoubleArray(self.toCancel)
-        else:
-            self.namesEntry = None
-            self.idsEntry = None
-            self.cancelEntry = None
-
-    def updateTable(self):
-        if not (self.namesEntry is not None and self.cancelEntry is not None):
-            return
-
-        # Get the commands to cancel
-        self.toCancel = self.cancelEntry.getDoubleArray([])
-        if self.toCancel:
-            for command in self.commandTable:
-                if id(command) in self.toCancel:
-                    command.cancel()
-            self.toCancel = []
-            self.cancelEntry.setDoubleArray(toCancel)
-
-        if self.runningCommandsChanged:
-            self.commands.clear()
-            self.ids.clear()
-            # Set the the running commands
-            for command in self.commandTable:
-                self.commands.append(command.getName())
-                self.ids.append(id(command))
-            self.namesEntry.setStringArray(self.commands)
-            self.idsEntry.setDoubleArray(self.ids)
-
-    def getSmartDashboardType(self):
-        return "Scheduler"
+    def _updateTable(self):
+        if self.namesEntry is not None and self.idsEntry is not None and self.cancelEntry is not None:
+            toCancel = self.cancelEntry.getDoubleArray([])
+            if toCancel:
+                for command in list(self.commandTable):
```

Comment:
Hm. I'm guessing the `list()` is required because `self.commandTable` is indirectly mutated by the loop? Should have a comment here to that effect.

If it's not, then leave off the `list()`.